src/V3Force.cpp

```diff
@@ -48,97 +53,116 @@ VL_DEFINE_DEBUG_FUNCTIONS;
 //######################################################################
 // Convert force/release statements and signals marked 'forceable'
 
-class ForceConvertVisitor final : public VNVisitor {
-    // TYPES
-    struct ForceComponentsVar final {
-        AstVar* const m_rdVarp;  // New variable to replace read references with
-        AstVar* const m_valVarp;  // Forced value
-        AstVar* const m_enVarp;  // Force enabled signal
-        explicit ForceComponentsVar(AstVar* varp)
-            : m_rdVarp{new AstVar{varp->fileline(), VVarType::WIRE, varp->name() + "__VforceRd",
-                                  varp->dtypep()}}
-            , m_valVarp{new AstVar{varp->fileline(), VVarType::VAR, varp->name() + "__VforceVal",
-                                   varp->dtypep()}}
-            , m_enVarp{new AstVar{varp->fileline(), VVarType::VAR, varp->name() + "__VforceEn",
-                                  (isRangedDType(varp) ? varp->dtypep() : varp->findBitDType())}} {
-            m_rdVarp->addNext(m_enVarp);
-            m_rdVarp->addNext(m_valVarp);
-            varp->addNextHere(m_rdVarp);
+static bool isRangedDType(AstNode* nodep) {
+    // If ranged we need a multibit enable to support bit-by-bit part-select forces,
+    // otherwise forcing a real or other opaque dtype and need a single bit enable.
+    const AstBasicDType* const basicp = nodep->dtypep()->skipRefp()->basicp();
+    return basicp && basicp->isRanged();
+}
+static bool isNotReplaceable(const AstVarRef* const nodep) { return nodep->user2(); }
+static void markNonReplaceable(AstVarRef* const nodep) { nodep->user2SetOnce(); }
```

Comment:
Avoid name pollution (even if static).  

```suggestion
static bool forceIsRangedDType(AstNode* nodep) {
    // If ranged we need a multibit enable to support bit-by-bit part-select forces,
    // otherwise forcing a real or other opaque dtype and need a single bit enable.
    const AstBasicDType* const basicp = nodep->dtypep()->skipRefp()->basicp();
    return basicp && basicp->isRanged();
}
static bool forceIsNotReplaceable(const AstVarRef* const nodep) { return nodep->user2(); }
static void forceMarkNonReplaceable(AstVarRef* const nodep) { nodep->user2SetOnce(); }
```